src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -668,52 +662,40 @@ def format_length(self):
         return len(str(self.size))
     counter = Counter(len(invalid_vts))
 
-    def check_cache(vts):
-      """Manually checks the artifact cache (usually immediately before compilation.)
-
-      Returns true if the cache was hit successfully, indicating that no compilation is necessary.
-      """
-      if not self.artifact_cache_reads_enabled():
-        return False
-      cached_vts, _, _ = self.check_artifact_cache([vts])
-      if not cached_vts:
-        self.context.log.debug('Missed cache during double check for {}'
-                               .format(vts.target.address.spec))
-        return False
-      assert cached_vts == [vts], (
-          'Cache returned unexpected target: {} vs {}'.format(cached_vts, [vts])
-      )
-      self.context.log.info('Hit cache during double check for {}'.format(vts.target.address.spec))
-      counter()
-      return True
+    jobs = []
+    invalid_target_set = set(invalid_targets)
+    for ivts in invalid_vts:
+      # Invalidated targets are a subset of relevant targets: get the context for this one.
+      compile_target = ivts.target
+      invalid_dependencies = self._collect_invalid_compile_dependencies(compile_target,
+                                                                        invalid_target_set)
 
-    def should_compile_incrementally(vts, ctx):
-      """Check to see if the compile should try to re-use the existing analysis.
+      jobs.extend(
+        self.create_compile_jobs(
+          compile_target,
+          compile_contexts,
+          invalid_dependencies,
+          ivts,
+          counter,
+          classpath_product_key))
+    return jobs
 
-      Returns true if we should try to compile the target incrementally.
-      """
-      if not vts.is_incremental:
-        return False
-      if not self._clear_invalid_analysis:
-        return True
-      return os.path.exists(ctx.analysis_file)
+  def create_compile_jobs(self, compile_target, all_compile_contexts, invalid_dependencies, ivts,
```

Comment:
Is this now intended for overriding? If so, could you add a comment explaining the shape of the return value (I'm less worried about the parameters).

Reply:
Sure.